src/ORM/Search/SearchContext.php

```diff
@@ -104,7 +105,18 @@ public function __construct($modelClass, $fields = null, $filters = null)
      */
     public function getSearchFields()
     {
-        return ($this->fields) ? $this->fields : singleton($this->modelClass)->scaffoldSearchFields();
+        if ($this->fields->exists()) {
```

Comment:
It's not safe to assume `$this->fields` is not null because `setField` below doesn't guard against a `null` value.
```suggestion
        if ($this->fields && $this->fields->exists()) {
```

Reply:
Done, but used the [null safe operator](https://www.php.net/manual/en/language.oop5.basic.php#language.oop5.basic.nullsafe) instead.

Reply:
For safety, I've done this in the other places that don't currently have null protection for `$this->fields` as well.